File: dag_1/behandling.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

int main() {
    
    int sum = 0;
    int maxsum = 0;
    int maxsum2 = 0;
    int maxsum3 = 0;

    fstream data;

    data.open("input.txt",ios::in); //open a file to perform read operation using file object
   if (data.is_open()){ //checking whether the file is open
      string bd;
      while(getline(data, bd)){ //read data from file object and put it into string.
         cout << bd << " "; //print the data of the string

         if (bd == "") {
            printf("\n");
            if (sum>maxsum) {maxsum3=maxsum2; maxsum2=maxsum; maxsum=sum;}
            else if (sum>maxsum2 && sum<=maxsum) {maxsum3=maxsum2; maxsum2=sum;}
            else if (sum>maxsum3 && sum<=maxsum2) {maxsum3=sum;}
            index+=1;
            sum=0;
         } else {
            sum=sum+stoi(bd);
         }

      }
      cout<<"\nThe max sum is: "<<maxsum;
      cout<<"\nThe three max sums are: "<<maxsum<<" "<<maxsum2<<" "<<maxsum3<<"\n"<<"The total sum is: "<<maxsum+maxsum2+maxsum3<<"\n";
      }

    return 0;
}

File: dag_1_python/behandling.py
sum=0
max_sum=0
max_sum2=0
max_sum3=0

with open("input.txt") as input:
    data = [i for i in input.read().strip().split("\n")]
    for item in data:
        if item == '': 
            if sum>max_sum:
                max_sum2=max_sum
                max_sum=sum
            elif sum>max_sum2:
                max_sum3=max_sum2
                max_sum2=sum
            elif sum>max_sum3:
                max_sum3=sum
            sum=0
        else:
            sum+=int(item)
            
print("Den højeste sum af kalorier er:",max_sum)
print("Summen af de tre højeste mængder kalorier er: ",max_sum+max_sum2+max_sum3)